system/service_windows_test.go

```diff
@@ -0,0 +1,67 @@
+package system
+
+import (
+	"testing"
+
+	"github.com/aelsabbahy/goss/util"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+func cleanupTestSvc(t *testing.T, svcName string, m *mgr.Mgr) {
+	svc, err := m.OpenService(svcName)
+	if err != nil {
+		// We assume that the service does not exist.
+		return
+	}
+	if err = svc.Delete(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDetectServiceWinMgr(t *testing.T) {
+	m, err := mgr.Connect()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create and auto-cleanup our service.
+	const svcName = "randomGossTestService"
```

Comment:
May want to randomise the service name a little. It's possible under a number of conditions for a delete to succeed but only actually clear the service after a reboot, so if another service with the same name is added before a reboot, that will fail.

Reply:
Done, I've also removed that `cleanupTestSvc` procedure since we should not need it now.